diff -r 180ed0c2a6ab src/client/windows/build/common.gypi
--- a/src/client/windows/build/common.gypi	Wed Mar 27 04:16:25 2013 -0700
+++ b/src/client/windows/build/common.gypi	Wed Mar 27 05:28:49 2013 -0700
@@ -400,9 +400,8 @@
       'win_release_Optimization%': '2', # 2 = /Os
       'win_debug_Optimization%': '0',   # 0 = /Od
       # See http://msdn.microsoft.com/en-us/library/aa652367(VS.71).aspx
-      'win_release_RuntimeLibrary%': '0', # 0 = /MT (nondebug static)
-      'win_debug_RuntimeLibrary%': '1',   # 1 = /MTd (debug static)
-
+      'win_release_RuntimeLibrary%': '2', # 2 = /MT DLL
+      'win_debug_RuntimeLibrary%': '3',   # 3 = /MT DEbug DLL
       'release_extra_cflags%': '',
       'debug_extra_cflags%': '',
       'release_valgrind_build%': 0,
